### ISSUE TITLE
Object Detection on KITTI db - error code -11 / probably learning rate issue?

### ISSUE THREAD

**Issue by tamasbalassa** [NVIDIA/DIGITS#1631]
Hi,

My final aim is to train my own network that is able to find specific objects on microscopy images, but first I would like to see and learn the workflow on the KITTI db. I followed everything as it is here: https://github.com/NVIDIA/DIGITS/tree/master/examples/object-detection

But when I run this setup, at **random** epoch it stops with error -11. I have a 4GB GTX 960m GPU, so I set the Batch size to 2 and the accumulation to 5.

The strange fact is that the training can be done easily even with 50 epochs if I set the learning rate to 0,01, but as it is expected the mAP stays on 0 (no converging / cant recognize the cars). When I set back to 0,0001 (or even 0,001), than the error comes again.

From the following output files (error at 10th and 16th epoch), first my thought was that there is something wrong with CV, but how can the training be done without any problem when the LR is higher? This makes me feel, that everything is fine with CV...
[caffe_output_epoch10.txt](https://github.com/NVIDIA/DIGITS/files/985992/caffe_output_epoch10.txt)
[caffe_output_epoch16.txt](https://github.com/NVIDIA/DIGITS/files/985993/caffe_output_epoch16.txt)

I stuck here since two weeks, I would appreciate some comments or tips, what or how to improve the training process.

**Comment by ontheway16** [comment]
Hi, I do not know meaning of error 11 but lets check a few things. Did you set input images' resolutions to 1248x384? Also, do you have one (ie, car) or more than one class in label files? Did you change original class names? (Car, pedestrian,  Van, etc...)

**Comment by tamasbalassa** [comment]
Hi @ontheway16 and thank You for your answer!

> Did you set input images' resolutions to 1248x384?
 - Yes, when I created the dataset. (screenshot attached)
![kitti_db](https://cloud.githubusercontent.com/assets/3785204/25847882/384c7dc8-34b8-11e7-88f8-73fd009ffd9e.png)




> do you have one (ie, car) or more than one class in label files?
- Yes, it is from the original KITTI dataset. (see the attached file for example)
[000004.txt](https://github.com/NVIDIA/DIGITS/files/986395/000004.txt)

> Did you change original class names? (Car, pedestrian, Van, etc...)
- No, ofc I did not changed them.

I am also attaching a screenshot about the details of my training setup:
![training_details](https://cloud.githubusercontent.com/assets/3785204/25847964/9b220710-34b8-11e7-96c1-8a5ec6464c0b.png)

**Comment by ontheway16** [comment]
Everything seems in place.. Did you try other examples, like object recognition etc? (mnist dataset). Are they working without issues?
Randomly occuring errors also brings hardware related issues into mind.. In case of suspicion, you can monitor temp/GPU_load with $nvidia-settings

Thats all I can think of, at the moment.

**Comment by tamasbalassa** [comment]
> Did you try other examples, like object recognition etc? (mnist dataset)
- Yes, I tried Classification on mnist and worked perfectly!

I was monitoring everything meanwhile, the GPU temp was between 60-65°C and the utilization between 90-100%. 

But I am pointing back again, I tried 2 times with 0,01 LR and 50 epochs - and no errors occurred.

**Comment by ontheway16** [comment]
As a last resort, try set batch & accumulation to 1 and run again.

**Comment by tamasbalassa** [comment]
That was also my first idea when I got these errors, but didn't worked.

But today, just double check (and I was also pretty curious about this), set them to 1 and started 3 training (not at the same time, but with exactly the same setup) and all of them failed at the same epoch with the same error message:  
`Iteration 6368 (4.56143 iter/s, 174.507s/796 iter), loss = 26.9224`
`Train net output #0: loss_bbox = 6.43325 (* 2 = 12.8665 loss)`
`Train net output #1: loss_coverage = 14.0558 (* 1 = 14.0558 loss)`
`Iteration 6368, lr = 0.0001`
`Snapshotting to binary proto file snapshot_iter_6373.caffemodel`
`Snapshotting solver state to binary proto file snapshot_iter_6373.solverstate`
`Iteration 6373, Testing net (#0)`
`Ignoring source layer train_data`
`Ignoring source layer train_label`
`Ignoring source layer train_transform`

![failedtraining](https://cloud.githubusercontent.com/assets/3785204/25893299/87b6007c-3577-11e7-8871-b3f515634d6e.png)

So I think I was wrong and there is no "random" crash, but there should be an explicable reason for this.

**Comment by tamasbalassa** [comment]
I still don't know what was the problem. But I reinstalled my system and now I installed DIGITS in some other way than before and everything works perfectly. Thanks for your help!
